web/src/screens/DeepLinkScreen/DeepLinkScreen.tsx

```diff
@@ -0,0 +1,131 @@
+import { FC, useCallback, useEffect } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { alertPush, selectCurrentLanguage } from 'web/src/modules';
+import { useDispatch, useSelector } from 'react-redux';
+import { p2pUrl } from 'web/src/api';
+import {
+  DeeplinkResultParams,
+  DeeplinkType,
+  useActivateDeeplink,
+} from 'web/src/hooks/mutations/useActivateDeeplink';
+import { AdvertSingleSource, PaymethodSource } from 'web/src/modules/p2p/types';
+import { fetchWithCreds } from 'web/src/helpers/fetch';
+import { useUser } from 'web/src/components/app/AppContext';
+import { Spinner } from 'web/src/components/ui/Spinner';
+import { Container } from 'web/src/components/ui/Container';
+import { Box } from 'web/src/components/ui/Box';
+
+export default function generateAdvertLink({
+  advert,
+  paymethod,
+}: {
+  advert: AdvertSingleSource;
+  paymethod: PaymethodSource;
+}) {
+  const chpu = `${advert.type === 'selling' ? 'buy' : 'sell'}-${advert.cryptocurrency}-${
```

Comment:
Ещё небольшое изменение нужно
```suggestion
  const seoUrl = `${advert.type === 'selling' ? 'buy' : 'sell'}-${advert.cryptocurrency}-${
```

Reply:
done